resources/asciidoctor/spec/copy_images_spec.rb

```diff
@@ -29,23 +29,41 @@ def copy_attributes(copied)
     }
   end
 
-  spec_dir = File.dirname(__FILE__)
-
   ##
   # Like the 'convert' shared context, but also captures any images that
   # would be copied by the conversion process to the `convert` array. That
   # array contains tuples of the form
   # [image_path_from_asciidoc_file, image_path_on_disk] and is in the order
   # that the images source be copied.
   shared_context 'convert intercepting images' do
-    let(:assert_no_logs) { false }
+    # TODO: once we've switched all of the examples in this file we can probably
+    #  drop this.
     include_context 'convert with logs'
 
     # [] is the initial value but it is mutated by the conversion
-    let(:copied) { [].dup }
-    let(:convert_attributes) { copy_attributes(copied) }
+    let(:copied_storage) { [].dup }
```

Comment:
I don't follow why you are duping the Array?

Reply:
The array is mutated by the conversion process. We stick tuples for everything that we would copy into it instead. I think I'd like to replace it with a mock object eventually. Let me play around with not duping it and see if that breaks this.